src/pages/index.js

```diff
@@ -26,92 +52,196 @@ document.addEventListener("DOMContentLoaded", () => {
     formValidators[formId] = validator;
   });
 
-  // UserInfo instance
+  // API Client
+  const apiClient = new APIClient({ baseUrl: API_BASE_URL });
+
+  // User Info
   const userInfo = new UserInfo({
     nameSelector: ".content__title",
     descriptionSelector: ".content__description",
   });
-  // Image popup instance
+
+  // Fetch User Info
+  (async function fetchUserData() {
+    try {
+      const userData = await apiClient.get("/users/me", {
+        headers: { authorization: AUTH_TOKEN },
+      });
+      userInfo.setUserInfo({
+        name: userData.name,
+        description: userData.about,
+      });
+      profilePicture.src = userData.avatar;
+      profilePicture.alt = userData.name || "User Avatar";
+    } catch (error) {
+      console.error(`Error fetching user data: ${error.message}`);
+    }
+  })();
+
+  // Popups
   const imagePopup = new PopupWithImage(".modal_type_image");
   imagePopup.setEventListeners();
 
-  // Section instance
-  const section = new Section(
-    {
-      items: initialCards,
-      renderer: renderer,
-    },
-    CARD_LIST_SELECTOR
+  const deleteCardPopup = new PopupWithConfirm(".modal_type_confirm-delete");
+  deleteCardPopup.setEventListeners();
+
+  const updateAvatarPopupForm = new UserUpdateAvatarForm(
+    ".modal_type_update-avatar",
+    async (formData) => {
+      const avatar = formData.imageUrl?.trim();
+      try {
+        const updateUserAvatar = await apiClient.patch(
+          "/users/me/avatar",
+          {
+            avatar,
+          },
+          { headers: { authorization: AUTH_TOKEN } }
+        );
+        profilePicture.src = updateUserAvatar.avatar;
+        profilePicture.alt = "User Avatar";
```

Comment:
Only methods of the `UserInfo` class should set/get the profile data (including the **avatar**)